src/routes/volume/detail/Snapshots.js

```diff
@@ -45,23 +45,34 @@ class Snapshots extends React.Component {
         return
       }
       let actions
-      if (action.type === 'snapshotDelete') {
-        actions = [{
-          url: this.props.volume.actions.snapshotPurge,
-        }]
-      }
-      this.props.dispatch({
-        type: 'snapshotModal/snapshotAction',
-        payload: {
-          type: action.type,
-          actions,
-          url: this.props.volume.actions[action.type],
-          params: {
-            name: action.payload && action.payload.snapshot ? action.payload.snapshot.name : undefined,
+      if (!action.payload?.snapshot?.removed) {
```

Comment:
Do we need this condition? Because whatever the value of `removed` is, we always need to convert to use `snapshotPurge` with `snapshotDelete`.

Also, this chang will cause other actions instead of `snapshotBackup`, `toggleShowRemoved` and `backup` converted to `snapshotPurge` when `snapshot.removed` is true.

I feel we should keep the case as is. WDYT?

Reply:
Sorry that this code is really misleading. The normal delete snapshot actually executes the request snapshotDelete+snapshotPurge twice. The corresponding called function will determine whether the actions array still has the requested link, and if there is, the request will continue.

<img width="900" alt="截屏2022-04-22 下午4 10 30" src="https://user-images.githubusercontent.com/27733391/164646332-15243e42-8dcf-490a-a82f-1c86f3d3813d.png">

To delete a snapshot that has been marked as removed, simply execute snapshotPurge. So this judgement is still needed.

I do need to optimize this code to make my intentions more explicit. I will adjust this pr to WIP. I'll let you know when the optimization is complete. Thank you for your patience in checking. @innobead

Reply:
ah, thanks for the explanation. Waiting for the update then.